google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableDefinition.java

```diff
@@ -165,6 +167,8 @@ Table toPb() {
         return (T) ExternalTableDefinition.fromPb(tablePb);
       case "MODEL":
         return (T) ModelTableDefinition.fromPb(tablePb);
+      case "SNAPSHOT":
```

Comment:
This seems like you're missing an extra class you need in java due to the structure, e.g. a SnapshotTableDefinition.  SnapshotDefinition is just one message field on the table resource.

Reply:
Yes, I refactored the class to `SnapshotTableDefinition`. Much better this way. Thanks for noticing this!